### PR TITLE
[FAI-9751] Add copy dashboard to workspace UI

### PR DESCRIPTION
**Pull request by fedetoledo**
> [!IMPORTANT]
> If you are merging into master, please add either a `backport` or a `no-backport` label to this PR. You will not be able to merge until you do this step. Refer to the section [Do I need to backport this PR?](https://www.notion.so/metabase/Metabase-Branching-Strategy-6eb577d5f61142aa960a626d6bbdfeb3?pvs=4#89f80d6f17714a0198aeb66c0efd1b71) in the Metabase Branching Strategy document for more details.

> **Warning**
>
> If that is your first contribution to Metabase, please sign the [Contributor License Agreement](https://docs.google.com/a/metabase.com/forms/d/1oV38o7b9ONFSwuzwmERRMi9SYrhYeOrkbmNaq9pOJ_E/viewform) (unless it's a tiny documentation change). Also, if you're attempting to fix a translation issue, please submit your changes to our [POEditor project](https://poeditor.com/join/project/ynjQmwSsGh) instead of opening a PR.

Closes https://github.com/metabase/metabase/issues/[issue_number]
https://faros-ai.atlassian.net/browse/FAI-9751

### Description

Describe the overall approach and the problem being solved.

### How to verify

Describe the steps to verify that the changes are working as expected.

1. New question -> Sample Dataset -> ...
2. ...

### Demo

_Upload a demo video or before/after screenshots if sensible or remove the section_

### Checklist

- [ ] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: frontend/src/metabase/collections/components/BulkActions.jsx [hunk 1]
```diff
@@ -32,6 +32,20 @@ function BulkActions({
...
   const canArchive = selected.every(item => canArchiveItem(item, collection));
   const isVisible = selected.length > 0;
+  const selectedItemsIds = selected.map(item => item.id);
+
+  const onCopyToAnotherWorkspace = () => {
+    const messageData = {
+      pipelines: {
+        type: "DashboardTransfer",
+        payload: {
+          selectedDashboards: selectedItemsIds,
```
Comment (kinfaikan) [inline frontend/src/metabase/collections/components/BulkActions.jsx:42]: items may not necessarily be dashboards?

File: frontend/src/metabase/collections/components/BulkActions.jsx [hunk 2]
```diff
@@ -32,6 +32,20 @@ function BulkActions({
...
+          shouldStartTransferring: true,
```
Comment (kinfaikan) [inline frontend/src/metabase/collections/components/BulkActions.jsx:43]: are there scenarios that `shouldStartTransferring` is set to `false`?

File: frontend/src/metabase/collections/components/BulkActions.jsx [hunk 3]
```diff
@@ -71,6 +96,26 @@ function BulkActions({
...
+              {isCopyToWorkspaceEnabled && selectedDashboards.length > 0 && (
+                <CardSide>
+                  <Tooltip
+                    display={
+                      nonDashboardsSelected.length > 0 ? "block" : "none"
+                    }
+                    zIndex={99999}
+                    color="white"
+                    bg={"brand"}
+                    label="Only dashboards can be copied. Please deselect all the collections."
```
Comment (kinfaikan) [inline frontend/src/metabase/collections/components/BulkActions.jsx:108]: nit: not sure if "collections" the right term in this context. maybe "Please deselect all the other items."?

File: frontend/src/metabase/collections/components/BulkActions.jsx [hunk 4]
```diff
@@ -32,6 +34,29 @@ function BulkActions({
...
+    state => state.embed.options.enable_copy_to_workspace,
+  );
+
+  const onCopyToAnotherWorkspace = () => {
+    const messageData = {
+      pipelines: {
+        type: "DashboardTransfer",
+        payload: {
+          selectedDashboards,
+          copyButtonClicked: true,
```
Comment (kinfaikan) [inline frontend/src/metabase/collections/components/BulkActions.jsx:54]: nit: is the `copyButtonClicked" attribute truly needed?

File: frontend/src/metabase/collections/components/BulkActions.jsx (hunk 1 above)
Comment (fedetoledo) [reply frontend/src/metabase/collections/components/BulkActions.jsx]: Items may be collections, which are not allowed to copy. Updated

File: frontend/src/metabase/collections/components/BulkActions.jsx (hunk 2 above)
Comment (fedetoledo) [reply frontend/src/metabase/collections/components/BulkActions.jsx]: No, updated to buttonClicked flag

File: frontend/src/metabase/collections/components/BulkActions.jsx (hunk 4 above)
Comment (fedetoledo) [reply frontend/src/metabase/collections/components/BulkActions.jsx:54]: you're right, not needed

**Review by kinfaikan — APPROVED** [review]

**Pull request merged**